src/dmd/mars.d

```diff
@@ -848,21 +848,107 @@ private int tryMain(size_t argc, const(char)** argv)
         fatal();
 
     // inlineScan incrementally run semantic3 of each expanded functions.
-    // So deps file generation should be moved after the inlinig stage.
-    if (global.params.moduleDeps)
+    // So deps file generation should be moved after the inlining stage.
+    if (global.params.moduleDeps || global.params.doMakeDeps)
     {
         foreach (i; 1 .. modules[0].aimports.dim)
             semantic3OnDependencies(modules[0].aimports[i]);
 
-        OutBuffer* ob = global.params.moduleDeps;
-        if (global.params.moduleDepsFile)
+        if (global.params.moduleDeps)
         {
-            auto deps = File(global.params.moduleDepsFile);
-            deps.setbuffer(cast(void*)ob.data, ob.offset);
-            writeFile(Loc.initial, &deps);
+            OutBuffer* ob = global.params.moduleDeps;
+            if (global.params.moduleDepsFile)
+            {
+                auto deps = File(global.params.moduleDepsFile);
+                deps.setbuffer(cast(void*)ob.data, ob.offset);
+                writeFile(Loc.initial, &deps);
+            }
+            else
+                printf("%.*s", cast(int)ob.offset, ob.data);
+        }
+        if (global.params.doMakeDeps)
+        {
+            // Recursive dependencies scanner, fills string tables with module source paths and string-imported files paths
+            void recursivelyAddModuleDependencies(Module m, StringTable* moduleImportedPaths, StringTable* stringImportedPaths)
+            {
+                auto mpath = m.srcfile.toString();
+                // Skip modules already in the table, add self
+                if (moduleImportedPaths.insert(mpath.ptr, mpath.length, null) is null)
```

Comment:
Please add an overload to `insert` that takes `const(char)[]` instead of a pointer and length.